File: src/main/java/com/dsm/pick/domains/service/NoticeService.java
package com.dsm.pick.domains.service;

import com.dsm.pick.domains.repository.NoticeRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class NoticeService {

    private final NoticeRepository noticeRepository;

    @Autowired
    public NoticeService(NoticeRepository noticeRepository) {
        this.noticeRepository = noticeRepository;
    }

    public List<String> getClubNotice() {
        return noticeRepository.findByDate(LocalDateTime.now(), "club");
    }

    public List<String> getMemberNotice() {
        return noticeRepository.findByDate(LocalDateTime.now(), "member");
    }
}

File: src/main/kotlin/com/dsm/pick/service/NoticeService.kt
package com.dsm.pick.service

import com.dsm.pick.controller.response.NoticeResponse
import com.dsm.pick.domain.attribute.Category
import com.dsm.pick.repository.NoticeRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.sql.Timestamp

@Service
@Transactional(readOnly = true)
class NoticeService(
    private val noticeRepository: NoticeRepository,
) {

    fun findNotice() =
        NoticeResponse(
            clubNotice = noticeRepository.findByCategoryAndDateAfter(
                Category.CLUB,
                Timestamp(System.currentTimeMillis() - (1000 * 60 * 60 * 24 * 7))
            ).map { it.content },
            memberNotice = noticeRepository.findByCategoryAndDateAfter(
                Category.MEMBER,
                Timestamp(System.currentTimeMillis() - (1000 * 60 * 60 * 24 * 7))
            ).map { it.content },
        )
}